Updated the workshop to .NET 8 and Hot Chocolate 14

glen-84 opened the pull request:
⚠️ WIP

- [x] README.md
- [x] Creating a new GraphQL server project.
- [x] [Remove] ~~Controlling nullability~~.
- [x] Understanding ~~GraphQL query execution and~~ DataLoader.
- [x] GraphQL schema design approaches.
- [x] Understanding middleware.
- [x] Adding complex filter capabilities.
- [x] Adding real-time functionality with subscriptions.
- [x] Testing the GraphQL server.
- [ ] (Dev-ops)

---

**To-do**

- Append stable/keyset order when using `[UseSorting]`.
- [Michael] Dev-ops.

glen-84 committed:
Updated the workshop to .NET 8 and Hot Chocolate 14

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/signed)](https://cla-assistant.io/ChilliCream/graphql-workshop?pullRequest=112) <br/>All committers have signed the CLA.

michaelstaib commented on code/session-1/GraphQL/Migrations/20240807140835_Initial.cs:
```diff
@@ -0,0 +1,37 @@
+#nullable disable
+
+namespace ConferencePlanner.GraphQL.Migrations
+{
```
can you use the namespace declaration without the curly braces?

glen-84 replied:
This is a generated file (at least, initially). I'm not sure if it makes sense to instruct the reader to make this change.

See also: https://github.com/dotnet/efcore/issues/14545.

michaelstaib commented:
Looks good so far

glen-84 committed:
Added session 3

glen-84 committed:
Added session 4

glen-84 committed:
Added session 5

glen-84 committed:
Added session 6

glen-84 committed:
Added session 7

glen-84 committed:
Added session 8

glen-84 committed:
Removed session 2

glen-84 committed:
Removed first item in ToCs

glen-84 committed:
Removed unused images

glen-84 committed:
Removed `complete` directory

glen-84 committed:
Made most properties init-only

glen-84 committed:
Updated dependencies

glen-84 committed:
Updated dependencies

glen-84 committed:
Updated schema snapshot

glen-84 committed:
Removed trackByName/tracksByName

glen-84 committed:
Updated screenshots

glen-84 committed:
Removed unused using directives

glen-84 committed:
Updated diagrams

glen-84 committed:
Updated dependencies

glen-84 committed:
Switched to using the [BindMember] attribute

glen-84 committed:
Added solution that includes all projects

glen-84 committed:
Optimized DataLoaders

glen-84 committed:
Used DataLoader interfaces

glen-84 committed:
Added AsNoTracking

glen-84 committed:
Updated dependencies

glen-84 committed:
Restructured session 1 to ensure that the AddGraphQLTypes method exists

glen-84 committed:
Updated dependencies

glen-84 committed:
Updated dependencies

glen-84 committed:
Added typeName argument for track ID

glen-84 committed:
Updated schema snapshot

glen-84 committed:
Added keyset and DataLoader pagination

glen-84 committed:
Updated dependencies

glen-84 committed:
Used the generic version of the ID descriptor method

glen-84 committed:
Updated dependencies

pwernicke committed:
Fix mismatched curly brackets in 3-schema-design.md SessionDataLoader… (#114)

Fix mismatched curly brackets in 3-schema-design.md SessionDataLoaders code snippet

The class-closing bracket occurred after the first method leaving the remaining methods outside of the scope of a class. This moves the class-closing bracket to the end of the code snippet.

glen-84 committed:
Updated dependencies

glen-84 committed:
Fixed grammatical error

glen-84 committed:
Corrected link

glen-84 committed:
Renamed Banana Cake Pop to Nitro

glen-84 committed:
Added DataLoader projections (#115)

glen-84 committed:
Updated dependencies

glen-84 committed:
Updated dependencies

glen-84 committed:
Updated dependencies

glen-84 committed:
Updated dependencies

glen-84 committed:
Updated dependencies

glen-84 committed:
Updated dependencies

glen-84 committed:
Updated dependencies

glen-84 committed:
Updated dependencies

glen-84 committed:
Updated dependencies

glen-84 committed:
Update dependencies
